### ISSUE TITLE
position_stack() should stack +ive & -ve values separately

### ISSUE THREAD

**Issue by hadley** [hadley/ggplot2#1691]
```R
library(ggplot2)
library(tibble)

df <- frame_data(
  ~x, ~g, ~y,
  1,  1,  1,
  1,  2,  -1,
  1,  3,  1,
  2,  1,  2,
  2,  2,  -3
)

ggplot(df, aes(x, y, fill= factor(g))) + 
  geom_bar(stat = "identity")
```

**Comment by thomasp85** [comment]
There are a lot of case where, even if supported, negative values doesn't make sense when stacking. As I see it it only makes sense for stacked bar charts - can you come with other examples...

I'm asking because I need to know which types of checks to make and warnings to throw, and also where it should be solved. It could be handled directly by `pos_stack()` but this function is also called by `pos_fill()` where it doesn't make sense to have negative values. Alternatively (and easier) the data could be split into two calls to `collide` in `PositionStack$compute_panel()` which would make the functionality specific to `PositionStack` (though it would still be a mess if coupled with `geom_area`

**Comment by hadley** [comment]
Can you come with an example where stacking makes sense apart from bars/areas? I think that's the only case we need to worry about, and it's reasonable for positive values to stack upwards from the y-axis, and negative values to stack downwards.

Your fix seems reasonable to me - I don't think there's any way to have an elegant solution without completely rewriting the stacking (which we don't want to do)

**Comment by thomasp85** [comment]
I cannot - which was why I asked whether you knew of any other use cases. I'll move forward with #1693

**Comment by smouksassi** [comment]
I think we can also stack dots which can be useful in some cases. http://www.lispworks.com/success-stories/inspiredata.html

http://docs.ggplot2.org/current/geom_dotplot.html

**Comment by thomasp85** [comment]
@smouksassi true. Dot plots should mainly be used for count data though, which makes them unsuitable to show negative values.

**Comment by smouksassi** [comment]
I was thinking about a kind of back to back dotplot like this:   https://nzmaths.co.nz/resource/fully-stretched:

**Comment by hadley** [comment]
Dot plots have to use a completely different algorithm so aren't relevant to this discussion.

**Comment by wmurphyrd** [comment]
Here's an example where the current behavior of position_stack for a mixture of positive and negative values produced the desired result: the total extent of the bar aligns with the sum of the values.
![image](https://cloud.githubusercontent.com/assets/10034859/17420143/aec2243a-5a66-11e6-987e-45750745968c.png). However, I did zoom-hide the negative side because it's not clear what's going on. (From https://github.com/wmurphyrd/BernieTax )

In another example, I wanted to show what was happening on the negative side with a dodge: 
![image](https://cloud.githubusercontent.com/assets/10034859/17420273/70a881fc-5a67-11e6-8add-6122fd7e6468.png). Although, in the end I used geom_tile with position_dodge after calculating the offsets manually
